samples/csharp_dotnetcore/49.qnamaker-all-features/BotServices.cs

```diff
@@ -34,5 +34,19 @@ private static string GetHostname(string hostname)
 
             return hostname;
         }
+
+        private static string GetEndpointKey(IConfiguration configuration)
+        {
+            var endpointKey = configuration["QnAEndpointKey"];
+
+            if(string.IsNullOrWhiteSpace(endpointKey))
```

Comment:
This is a sample - we don't need back-compat in a sample. 

The sample should work "as-is" and is a teaching tool.

Reply:
@cleemullins 
this 49.qnamaker-all-features sample is copied as is for "azure bot service"  default "createbot" template.

post this change merged into "azure bot service" template repo, "Azure Bot Service" will make the web app config change to use "QnAEndpointKey". But, this default "QnAAuthkey" will maintain backward compact.

This is requirement from docs to keep app setting name consistent with "QnAEndpointKey".

Reply:
Resolved by adding a comment in the code that articulates why this is needed.